### DISCUSSION TITLE
Including and excluding properties when using the collect() method?

### DISCUSSION THREAD

**Discussion by jonjakoblich** [spatie/laravel-data#791] [Q&A] [answered]
How can I include a Lazy property on when creating a collection of data objects?

I understand from the docs that this works (assume property 'name' is Lazy).

`SongData::from(Song::first())->include('name');`

But, because this returns a Collection, the following code will not work.

`SongData::collect(Song::all())->include('name');`

What ae my options for including Lazy properties with collections? I am working with v4. Thanks!

**Comment by rubenvanassche** [comment, marked as answer]
Since collect outputs the same type as it gets fed, the following:

> SongData::collect(Song::all())->include('name');

Will return an Eloquent collection which doesn't support includes. Try the following:

```php
SongData::collect(Song::all(), DataCollection::class)->include('name');
```

**Reply by jonjakoblich** [reply]
Thank you! Would this be the same for only() and except() also? Do the properties have to be Lazy for only() and except() to work?

**Reply by rubenvanassche** [reply]
Jup same thing, they don't have to be lazy and get executed before we evaluate lazy.
